shapely/geometry/base.py

```diff
@@ -60,34 +62,20 @@ def dump_coords(geom):
         raise ValueError('Unhandled geometry type: ' + repr(geom.type))
 
 
-def geometry_type_name(g):
-    if g is None:
-        raise ValueError("Null geometry has no type")
-    return GEOMETRY_TYPES[lgeos.GEOSGeomTypeId(g)]
+class _DummyGeometry:
+    # pygeos.from_shapely expects a python object that exposes
+    # the GEOS pointer as a __geom__ attribute
+
+    def __init__(self, g):
+        self.__geom__ = g
 
 
 def geom_factory(g, parent=None):
     # Abstract geometry factory for use with topological methods below
     if not g:
         raise ValueError("No Shapely geometry can be created from null value")
-    ob = BaseGeometry()
-    geom_type = geometry_type_name(g)
-    # TODO: check cost of dynamic import by profiling
-    mod = __import__(
-        'shapely.geometry',
-        globals(),
-        locals(),
-        [geom_type],
-        )
-    ob.__class__ = getattr(mod, geom_type)
-    ob._geom = g
-    ob.__p__ = parent
-    if lgeos.methods['has_z'](g):
-        ob._ndim = 3
-    else:
-        ob._ndim = 2
-    ob._is_empty = False
-    return ob
+    ob = _DummyGeometry(g)
```

Comment:
This seems backwards to me.
In my opinion we should never pass these pointers around as python `int` objects. Do you think we can remove the `geom_factory` completely from shapely 2.0?

Reply:
Yes, this is certainly only temporary. I don't refactor the full of Shapely in a single PR, and thus in other parts of shapely, `geom_factory` is still used together with ctypes. So those changes here with the DummyGeometry hack is just to ensure `geom_factory` keeps working. But the end goal of Shapely 2.0 for sure means that all usage of `geom_factory` is eliminated.